#160283233 set up axios instance

tutugodfrey opened the pull request:
#### What does this PR do?
- Add axios for making API requests

#### Description of Task to be completed?
- Install axios as project dependencies
- Set up and axios instance
- Add a function for making API requests using axios

#### How should this be manually tested?
- Clone the repository and in your terminal `npm install` to install dependencies, then run `npm run dev` to start the application in development mode
- When the browser open try clicking the button `Secondary` and check the console for messages

#### Any background context you want to provide?
- Two functions are defined one is for starting an axios instances which can be use out of the box
- The function apiCalls is accepts the axios instance, and other parameters (header method, body) and make an api call using axios

#### What are the relevant pivotal tracker stories?
#160283233

#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/30448913/45148086-628b5080-b1be-11e8-9a35-aaf9745b4926.png)
![image](https://user-images.githubusercontent.com/30448913/45153947-bea9a100-b1cd-11e8-9d23-3bde07bd70e4.png)


#### Questions:

rhenshaw56 commented on package.json:
```diff
@@ -20,14 +20,21 @@
+    "@babel/preset-react": "^7.0.0",
     "babel": "^6.23.0",
-    "babel-core": "^6.26.3",
+    "babel-core": "^7.0.0-bridge.0",
```
@tutugodfrey why the change ?

rhenshaw56 commented on src/App.js:
```diff
@@ -1,12 +1,22 @@
+   * using axios instance
+   * @returns { Object } response from server
+   */
+  makeRequest = async () => {
```
you can't do this, do it right, cos elint is flagging you

rhenshaw56 commented on src/services/requestHandler.js:
```diff
@@ -0,0 +1,73 @@
+ * @param { Object } body the request body
+ * @returns { instance } an instance of axios
+ */
+const axiosInstance = (headers, body) => {
```
add the default headers to the instance, we don't want to be passing in headers every time we need to make a request

rhenshaw56 commented on src/services/requestHandler.js:
```diff
@@ -0,0 +1,73 @@
+ * @returns { Object } response the response from the server
+ */
+const apiCalls = async (axiosInStance, method, path, headers, body) => {
+  /* eslint-disable no-console */
```
take all this out, you don't need it, it just complicates things

tutugodfrey replied:
Ok, I will attend to them.

andela-moe commented:
@tutugodfrey I am interested in knowing why the commit to the branch in this PR was done by Joseph Cobhams https://github.com/andela/rivendell-ah-client/pull/10/commits

@andela-moe the mixup is because my system is bad and I was given another system to use for the main time. The username reflect the username of the person formally in possession of the system. But I have updated the information now.

tutugodfrey committed:
feat(axios-instance): set up axios instance

- install axois
- define a function to start an axios instance
- define an example function for making api requests

[Delivers #160283233]

chidioguejiofor commented on src/services/requestHandler.js:
```diff
@@ -0,0 +1,22 @@
+ * @returns { instance } an instance of axios
+ */
+const axiosInstance = () => {
+  let apiUrl = 'https://authorhaven.herokuapp.com/api';
```
Instead of doing this. What do you think about using an environmental variable like process.env.BASE_URL
